### PR TITLE
Removes borg ability to welder themselves.

### PR DESCRIPTION
**Pull request by TerraGS**
:cl: Terra_GS (requested by imsxz)
balance: Removes the ability for cyborgs to weld themselves.
/:cl:

EDIT: People won't stop basing their argument over the placeholder reason I had here originally, so now I'm going to make it absolutely clear: Direct all non-code related questions to imsxz as that's where you will get the best answers.

### CODE REVIEW

**Comment by epicmemename** [comment]
+1, silicons having the ability to heal themselves + no combat slowdown makes them very annoying to deal with, also making engineering borg pretty much the only good one

**Comment by optimumtact** [comment]
>but in short borgs being able to weld themselves makes them incredibly resilient in any battle of attrition to the point that a few engiborgs is all it takes to kill a blob.

>to the point that a **few engiborgs** is all it takes to kill a blob.

Your stated reason for this nerf makes no sense, if there are a few engiborgs they can repair each other, they don't need to welder themself, given that this exact thing has been flipflop changed a couple of times now I'm going to need to see more of a justification than that

**Comment by GuyonBroadway** [comment]
Unless a human gets augmented, then they CAN fix themselves with a welder and get to be spaceproof too.

**Comment by TerraGS** [comment]
@optimumtact Again, it's imsxz you should be talking to, I just jotted something down so it wouldn't be left without a reason until imsxz came to clear things up.

**Comment by Tlaltecuhtli** [comment]
what about miner borgs they have a welder too and an actual reason to heal themselves

**Comment by DingleFlop** [comment]
I believe that this is needless. Borgs aren't particularly strong. They have a hard couple hard counters (flash and EMP) and things that have too much trouble dealing with them should gain a counter, rather than need the entirety of borgs.

**Comment by TerraGS** [comment]
@Tlaltecuhtli A future change to mining borgs to make them stronger in another way may be in order. This PR is meant to address a current issue.

**Comment by Daundra** [comment]
If you're having problems with engieborgs as a blob, the counter to that is to focus on becoming EMP type. This literally makes them completely irrelevant, easy to kill, and while you say imsxz was giving the real reason for the PR I think I should remind people that EMP blob is a thing and the whole borgs are the unstoppable blob counter thing is not entirely true. in fact, I think more people should consider engieborgs still cant fix burn damage. 90% of people's problems with engieborg self heal can be fixed with breaking into chemistry and making a SINGLE EMP grenade in less than 20 seconds.

As for the PR itself, the alternatives for engieborgs are to now futilely beg crewmembers to welder them / hope the cyborg rechargers were upgraded (they weren't) / hope the roboticst will do it (will, if they exist) / actually make engieborgs require the self repair module (not as bad in my opinion as the others but this still feels unnecessary)

the point is the engieborg (whose main purpose is to repair/maintain the station, and will probably be in EVA a lot,) being expected to rely on outside assistance in such human inhospitable environments will detract from their use to a serious degree, and god forbid your AI is the color rouge because if you cant repair yourself because 99% of the crew is validhunting the shit out of you well better hope you served your part by reinforcing the core before you die your sad death.

I will agree with imsxz in that the selfheal lets them isolate but I mean are any of the other borg modules expected to be in space/other dangerous areas? maybe they're SUPPOSED to be able to isolate themselves.

I however disagree with the fleshmend comparison. It's not instant, for one, and two, it's not gonna do anything for a borg's wiring. for real, I want to reiterate that engineer borgs cant fix their wires. just like all the other modules. let them have this one grace so they can keep sticking in EVA patching the hulls and fighting the carps etc. without having to crawl all the way back over a small booboo.

**Comment by obscolene** [comment]
@DingleFlop 
>I believe that this is needless. Borgs aren't particularly strong. They have a hard couple hard counters (flash and EMP) and things that have too much trouble dealing with them should gain a counter, rather than **nerf the entirety of borgs.**

Buddy, only engiborgs can repair themselves. Borgs can still repair other borgs perfectly fine.

**Comment by AlexTheSysop42** [comment]
Standard and mining can self-repair too

**Comment by PKPenguin321** [comment]
>captain fighting rogue borgs
>kill all of them but the engiborg
>every time i laser it it just skedaddles away with its no-damage-slowdown
>reappears 30 seconds later back at full health

👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍👍 maintainers, please add thsi!

**Comment by zxaber** [comment]
Do fleshmends get perma-stunned by a button that works at any distance?

There is a lockdown if the borgs need dealt with. You can cut the camera to prevent AI unlocking them, or just pull the board out and use it somewhere else. If the console was broken, you can print new boards in science.

If this *does* end up merged, can we at least get a buff to the rate of the self-repair upgrade, so that organics can later *give* the borgs the means to repair themselves in a decent amount of time?

**Comment by octareenroon91** [comment]
>every time i laser
>reappears 30 seconds later back at full health

how do you weld away fire damage again?

**Comment by PKPenguin321** [comment]
i dont know

**Comment by PKPenguin321** [comment]
@zxaber in the scenario mentioned the borgs were rogue and had already disposed of robotics consoles boards and it was REAaaaally lame

and to clarify much of the damage to the borg wasnt lasers, it was damage that they repaired by self welding, so dont bully me

**Comment by AlexTheSysop42** [comment]
>counters aren't relevant because they keep you from healing

what

If you want to kill a borg, use one of the counters. Otherwise there's much more they can do than self-heal. In PK's scenario, he could've used a flash or printed a console board to kill it easily.

**Comment by PKPenguin321** [comment]
>printed a console board 

No i couldn't

**Comment by AlexTheSysop42** [comment]
Why not then

**Comment by PKPenguin321** [comment]
now we're getting off topic but there were no mats and there was no research done

**Comment by zxaber** [comment]
@PKPenguin321 You're right, this is a bit off topic, but counterpoint; the board requires 500 glass (quarter of a sheet) and is available at round start.

The real point is, borgs still have counters that they can't fix themselves. Burn damage (EMP grenades and rifle), lockdown, or just grab a flash and some sunglasses and flash and smash. Self-welding is really useful out of combat, as engineer borgs often get pretty banged up just from doing their normal job.

**Comment by PKPenguin321** [comment]
if you absolutely cannot have a human weld you (like youre rogue and the crew knows it), have a second borg heal you, plain as that. if youre not rogue i cant think of any reason you couldnt just ask the crew for help

**Comment by zxaber** [comment]
The other day there was an event where we had a ton of grav anomalies spawning everywhere, and I was an engi borg handling it. Every single one took out like half my health because I had to get within one tile to scan it. The only reason I was able to sort of stem the tide of them was because I could repair myself.

Sometimes things are breaking and you're doing everything you can to protect the crew. Having to run all the way back to robo over every dent would make things much harder.

**Comment by octareenroon91** [comment]
>give the engiborg an emergency tools dispenser

**Comment by PKPenguin321** [comment]
>Having to run all the way back to robo over every dent would make things much harder.

Ok and? That's kind of the point

**Comment by GuyonBroadway** [comment]
>Ok and? That's kind of the point

Its really not fun though...

**Comment by AlexTheSysop42** [comment]
One time I was an engiborg trying to stop a traitor CE from delamming the SM. I would've died MANY times over if I couldn't leave and heal myself, and if I had to go to robotics it would've exploded by the time I got back.

**Comment by Dorsisdwarf** [comment]
@PKPenguin321 that argument could just as well apply to making humans unable to self-heal though, bad argument

**Comment by zombiegums** [comment]
Yeah, don't add this please. Just as I wouldn't run into a nuke op with just a toolbox, don't try to engage a borg unless you've got a flash or something.  Flashes are so cheap to make (750 metal, 750 glass) and you can kill a borg, or at least disable all of their modules with just 2 uses from one flash. If I am trying to stop someone from harming and then see a flash appear in their hand, I become incredibly wary because if I get hit once, I'm probably screwed.

Regarding blobs, I don't see how nerfing borgs in this way would stop them from being a nuisance. When I fight blobs as a borg, I can assure you that most borg players will heal each other because it's instant, unlike the self-heal. I also find it hard to see how this is any different from blobbernauts attacking targets and then retreating back to their node (or whatever it's called) to heal up. 

Ignore antag related qualms, this is also a pain in the arse for borgs that are trying to do their roles. Mining borgs have a dinky welding tool that is clearly meant for self-healing. Removing that would be awful.

**Comment by Arkvold** [comment]
Leave self-welding in.  It forces antags to actually be wary of engie and mining borgs and utilize fire damage and flashes when dealing with them.  _Welders cannot repair fire damage,_ only brute, and their welders have a limited supply of fuel as well.  Antags that know borgs are on the station should seriously consider breaking into the armory, into the RD's office, or the AI chamber to do something about them.

Borgs are already intensely limited compared to fleshies, with a strict set of inviolable laws, a sword of damocles hanging over their heads in the form of the RD's console and the AI, a limited set of tools, no ability to change roles without outside help, a limited supply of power, vulnerability to EMP, and no ability to interact with items outside of specific cases (such as an engie borg picking up floor tiles or metal to resupply their stocks.)  Their complete access, inherent spaceworthiness, invulnerability to toxins, and _ease of recovery_ is the upside to existing as a borg.

TL;DR - Borgs with welders have enough limitations to be balanced, and they present an interesting challenge to antags in that you can't just murderbone them with massive brute damage like you would everyone else, because they can scurry off and repair it, only to come back and harass you some more.  Switch up your tactics and stop relying on hulk hands or dswords, and actually use a laser, flash, flamer, or a harm-intent welder yourself.

**Comment by optimumtact** [comment]
I think I agree with the majority here, AOE flash stuns will wreck any group of borgs and immobilise them to be finished off.

It just means you have to ensure you make the borg dead, instead of damaging it down to crit and letting it scamper off.

**Comment by PKPenguin321** [comment]
AOE flashes dont stun borgs, or do ANYTHING to them, only melee flashes where you click directly on them

**Comment by ExcessiveUseOfCobblestone** [comment]
I was playing a blob round last night and a single engie borg could constantly just lolnope any progress and when I tried to hit them with all I had they just left clicked out of it after moving a tile or 2 back.

Utter cancer.

**Comment by optimumtact** [comment]
@ExcessiveUseOfCobblestone instead of whining like a baby about it, give blobs an EMP action purchaseable instead of trying to globally nerf borgs

**Comment by PKPenguin321** [comment]
"globally nerf" one of the several borg modules
